rfcs/2023-01-06-layout.md

```diff
@@ -0,0 +1,460 @@
+  paddingX="spacing.1"
+>
+  <Button>Button 1</Button>
+  <Button marginTop="spacing.1">Button 2</Button>
+  <Button marginTop="spacing.2">Button 3</Button>
+</Box>
+```
+
+Styling props will be the preferred choice for minor adjustments that can be done directly on the component without a need for creating extra `Box` wrapper _(which also keeps the DOM less nested and easier to read)_. Later on, we might want to expand the scope of these props to cover things beyond layout if need arises.
+
+As a tradeoff, styling props are pretty dangerous in terms of the flexibility they provide. They give power to customize any component to beyond recognition, for example you could customize the padding that comes on `Button`. In light of this capability we should support only a restricted set of properties to begin with _(which can't have adverse effect on look and feel of components)_.
+
+Styling props should be the preferred choice for minor adjustments only. You may note `padding*` is skipped since it can break existing padding of blade components. Supported restricted properties:
+
+| Category            | Props                                                                          | Notes                                                                                                                                                               |
+| ------------------- | ------------------------------------------------------------------------------ | ------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| Spacing             | `margin*`                                                                      | `marginTop`, `marginLeft`, `marginRight`, `marginBottom`, `marginX`, `marginY` (React Native `marginHorizontal`, `marginVertical` mapped to these)                  |
+| Flexbox             | `alignSelf`, `justifySelf`, `order`                                            | Only restricted set of properties applicable on flex children. Note: `justifySelf` doesn't work for flex layout but it can be used for block layout (web only).     |
+| Position            | `position`, `zIndex`, `top`, etc.                                              | [List of props](https://github.com/styled-system/styled-system/blob/master/docs/table.md#position) referenced from styled-system.                                   |
+| Grid **(web only)** | `gridColumn*`, `gridRow*`, `gridArea`, `justifySelf`, `alignSelf`, `placeSelf` | Only restricted set of properties applicable on [grid children](https://css-tricks.com/snippets/css/complete-guide-grid/#aa-properties-for-the-childrengrid-items). |
+| Layout              | None                                                                           |                                                                                                                                                                     |
```

Comment:
I would have also asked for float but I understand why you chose to omit it.

![time-touse-float-ijustbroke-the-wholesite-everytime-i-deal-with-44245421](https://user-images.githubusercontent.com/2913308/215750497-40b9c991-526d-44a7-9c33-e25aee4ac5f7.png)

Reply:
New proposal: A Clearfix component in blade. Let's remove all the flexbox stuff. 

```ts
<Clearfix />
```